models/stics/R/write.config.STICS.R

```diff
@@ -0,0 +1,405 @@
+  sols_list <- XML::xmlToList(sols_xml)
+  
+  sols_list$sol$.attrs[["nom"]] <- paste0("sol", defaults$pft$name)
+  
+  # DO NOTHING FOR NOW
+  
+  # write the tec file
+  XML::saveXML(PEcAn.settings::listToXml(sols_list, "sols"), 
+          file = file.path(rundir, "sols.xml"), 
+          prefix = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n')
+  
+  
+  ######################### Prepare Weather Station File ###############################
+  
+  ## this is where we modify more initial conditions and site characteristics
+  
+  # read in template sta file
+  sta_xml  <- XML::xmlParse(system.file("pecan_sta.xml", package = "PEcAn.STICS"))
+  sta_list <- XML::xmlToList(sta_xml)
+  
+  # change latitute
+  sta_list[[1]][[3]]$text <- settings$run$site$lat
+  
+  # DO NOTHING ELSE FOR NOW
+  
+  # Should these be prepared by met2model.STICS?
+  
+  # write the sta file
+  XML::saveXML(PEcAn.settings::listToXml(sta_list, "fichiersta"), 
+          file = file.path(rundir, paste0(tolower(sub(" .*", "", settings$run$site$name)), "_sta.xml")), 
+          prefix = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n')
+  
+
+  
+  
+  
+  ############################## Prepare LAI forcing ####################################
+  ## skipping for now
+  
+  
+  ############################## Param gen / newform ####################################
+  
+  ## DO NOTHING
+  gen_xml  <- XML::xmlParse(system.file("param_gen.xml", package = "PEcAn.STICS"))
+  gen_list <- XML::xmlToList(gen_xml)
+  XML::saveXML(PEcAn.settings::listToXml(gen_list, "fichierpar"), 
+               file = file.path(rundir, "param_gen.xml"), 
+               prefix = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n')
+  
+  newf_xml  <- XML::xmlParse(system.file("param_newform.xml", package = "PEcAn.STICS"))
+  newf_list <- XML::xmlToList(newf_xml)
+  XML::saveXML(PEcAn.settings::listToXml(newf_list, "fichierparamgen"), 
+               file = file.path(rundir, "param_newform.xml"), 
+               prefix = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n')
+
+  
+  ############################ Prepare Technical File ##################################
+  
+  ## this is where we modify management practices
+  
+  # read in template tec file
+  tec_xml  <- XML::xmlParse(system.file("pecan_tec.xml", package = "PEcAn.STICS"))
+  tec_list <- XML::xmlToList(tec_xml)
+  
+  # If harvest file is given, use given dates 
+  # this will need  more complicated checks and file formats
+  if(!is.null(settings$run$inputs$harvest)){
+    
+    h_days <- as.matrix(utils::read.table(settings$run$inputs$harvest$path, header = TRUE, sep = ","))
+    
+    # probably should use nicer list manipulation techniques, but these template files are static for now
+    # save last list to add at the end
+    attr_sublist <- tec_list[[8]][[1]][[1]][[2]][[2]][[1]][[4]] 
+    tec_list[[8]][[1]][[1]][[2]][[2]][[1]][[4]] <- NULL
```

Comment:
Ah yes, XML, the elegant format that programmers love

(I haven't looked at the files, but at refactor time this can probably be made into an xpath query that's still ugly but maybe more transparent. Feel free to ping me when revisiting)